tests/test_transaction.py

```diff
@@ -7,17 +7,21 @@
     "path",
     [
         SNOWBALL_WHEEL,
-        f"/{SNOWBALL_WHEEL}" f"a/{SNOWBALL_WHEEL}",
+        f"/{SNOWBALL_WHEEL}",
+        f"a/{SNOWBALL_WHEEL}",
         f"/a/{SNOWBALL_WHEEL}",
         f"//a/{SNOWBALL_WHEEL}",
     ],
 )
-@pytest.mark.parametrize("protocol", ["https:", "file:", "emfs:", ""])
+@pytest.mark.parametrize(
+    "protocol",
+    ["http:", "https:", "file:", "emfs:", ""],
+)
 def test_parse_wheel_url1(protocol, path):
     pytest.importorskip("packaging")
     from micropip.transaction import WheelInfo
 
-    url = protocol + path
+    url = protocol + path if protocol else "file:///" + path
```

Comment:
Why not put `file:///` in the parametrize decorator and do:
```suggestion
    url = protocol + path
```